pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -610,19 +666,69 @@ func addFinalizerToPVC(ctx context.Context, client client.Client, pvc *v1.Persis
 // from a given PersistentVolumeClaim.
 func removeFinalizerFromPVC(ctx context.Context, client client.Client, pvc *v1.PersistentVolumeClaim) error {
 	log := logger.GetLogger(ctx)
+	finalizerFound := false
 	for i, finalizer := range pvc.Finalizers {
 		if finalizer == cnsoperatortypes.CNSPvcFinalizer {
 			log.Debugf("Removing %q finalizer from PersistentVolumeClaim: %q on namespace: %q",
 				cnsoperatortypes.CNSPvcFinalizer, pvc.Name, pvc.Namespace)
 			pvc.Finalizers = append(pvc.Finalizers[:i], pvc.Finalizers[i+1:]...)
+			finalizerFound = true
+			break
 		}
 	}
-	err := client.Update(ctx, pvc)
+	if !finalizerFound {
+		log.Debugf("Finalizer: %q not found on PersistentVolumeClaim: %q on namespace: %q not found. Returning nil",
+			cnsoperatortypes.CNSPvcFinalizer, pvc.Name, pvc.Namespace)
+		return nil
+	}
+	err := updateSVPVC(ctx, client, pvc)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("PersistentVolumeClaim: %q on namespace: %q not found. Returning nil", pvc.Name, pvc.Namespace)
+			return nil
+		}
 		log.Errorf("failed to update PersistentVolumeClaim: %q on namespace: %q. Error: %+v",
 			pvc.Name, pvc.Namespace, err)
 	}
 	return err
+
+}
+
+func updateSVPVC(ctx context.Context, client client.Client,
+	pvc *v1.PersistentVolumeClaim) error {
+	log := logger.GetLogger(ctx)
+	err := client.Update(ctx, pvc)
+	if err != nil {
+		if apierrors.IsConflict(err) {
+			log.Infof("Observed conflict while updating the SV PVC %q in namespace %q."+
+				"Reapplying changes to the latest SV PVC object.", pvc.Name, pvc.Namespace)
+
+			// Fetch the latest pvc object from the API server and apply changes on top of it.
+			latestPVCObject := &v1.PersistentVolumeClaim{}
+			err = client.Get(ctx, k8stypes.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, latestPVCObject)
+			if err != nil {
+				log.Errorf("Error fetching the SV PVC with name: %q on namespace: %q. Err: %+v",
+					pvc.Name, pvc.Namespace, err)
+				return err
+			}
+
+			// The callers of updateSVPVC are only updating the instance finalizers
+			// Hence we copy the finalizers from the instance passed for update
+			// on the latest PVC object from API server.
+			latestPVCObject.Finalizers = pvc.Finalizers
```

Comment:
This will remove finalizers added by other controllers after `pvc.Finalizer` is initialized. Why don't you remove the finalizer you care about?

Reply:
Handling this using a bool. Please see the latest commit